salt/fileclient.py

```diff
@@ -980,15 +983,26 @@ def get_file(self,
                 else:
                     return False
             fn_ = salt.utils.fopen(dest, 'wb+')
-        while True:
+        # prevent endless loops
+        tries = 1000000
+
+        while tries > 0:
+            tries = tries  -1
+            if tries == 0:
+                raise Exception('failed to fetch file')
```

Comment:
This function should never raise exceptions because the higher level caller does not handle the exceptions and this would break a larger run and float the exception up to the user. stick with the failure case in the method of returning False so that the calling methods can handle it